.github/workflows/e2e.yaml

```diff
@@ -48,21 +51,27 @@ jobs:
             VERSION=${branch#"$prefix"}
           fi
           
-          if [ "${{ github.event.inputs.version }}" != "" ]; then
+          if [[ "${{ github.event.inputs.version }}" != "" ]]; then
```

Comment:
Let's pass inputs to shell actions through env vars and avoid interpolating them in the rendered shell scripts.

Reply:
I actually want them like this, having them inline is so much better for debugging, you can see in the step script in GH if got something wrong.

Reply:
This is a security best practice: https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#good-practices-for-mitigating-script-injection-attacks

And you can debug the values of environment variables in shell steps by clicking on the collapsed script, the env var values will be shown at the bottom, see this screenshot (`env: FLUX_VERSION: v2.4.0`):

![Screenshot from 2025-01-18 14-56-26](https://github.com/user-attachments/assets/171aecdd-c42e-4b8c-b072-ae66cf49323b)